flexmeasures/ui/crud/assets.py

```diff
@@ -149,16 +153,29 @@ class AssetCrudUI(FlaskView):
 
     @login_required
     def index(self, msg=""):
-        """/assets"""
-        get_assets_response = InternalApi().get(
-            url_for("AssetAPI:index"), query={"account_id": current_user.account_id}
-        )
-        assets = [
-            process_internal_api_response(ad, make_obj=True)
-            for ad in get_assets_response.json()
-        ]
+        """GET from /assets
+
+        List the user's assets. For admins, list across all accounts.
+        """
+        assets = []
+
+        def get_asset_by_account(account_id) -> List[GenericAsset]:
+            get_assets_response = InternalApi().get(
+                url_for("AssetAPI:index"), query={"account_id": account_id}
+            )
+            return [
+                process_internal_api_response(ad, make_obj=True)
+                for ad in get_assets_response.json()
+            ]
+
+        if user_has_admin_access(current_user, "read"):
+            for account in Account.query.all():
+                assets += get_asset_by_account(account.id)
```

Comment:
Two questions:
1. Do we not want to show public assets, too?
2. Why not query the asset table directly (saving \<number of accounts\> queries)? Optionally masking those without an owner (see question 1).

Reply:
1. I can add a parameter to also get `account=None`. I'd say that dashboard does not need it, but admins looking at the asset listing might be interested to find them there.
2. A design choice we made earlier is not to let the UI code call the DB directly. The API is the place which really decides what can be done. See also my other comments.